core/sdk-core/src/main/java/software/amazon/awssdk/core/client/config/ClientOverrideConfiguration.java

```diff
@@ -230,19 +234,30 @@ public ExecutionAttributes executionAttributes() {
         return executionAttributes;
     }
 
+    /**
+     * The request compression configuration object, which includes options to enable/disable request compression and set the
+     * minimum compression threshold.
+     *
+     * @see Builder#requestCompressionConfiguration(RequestCompressionConfiguration)
+     */
+    public RequestCompressionConfiguration requestCompressionConfiguration() {
```

Comment:
I think this class returns optional for unset properties, we should probably be consistent.